tests/integration/test_api.py

```diff
@@ -569,3 +571,130 @@ def test_prediction_collection_no_filter(client):
+    assert actual_items3[0]["barcode"] == "123"
+    assert actual_items3[0]["type"] == "category"
+    assert actual_items3[0]["value_tag"] == "en:eu-organic"
+    assert actual_items3[1]["id"] == prediction4.id
+
+    # test that we can filter "barcode", "value_tag", "keep_types" prediction
+    actual_prediction4 = get_predictions(
+        barcode="123", value_tag="en:eu-organic", keep_types=["category"]
+    )
+    actual_items4 = [item.to_dict() for item in actual_prediction4]
+    assert actual_items4[0]["id"] == prediction3.id
+    assert len(actual_items4) == 1
+
+    # test to filter results with "label" and "category" prediction
+    actual_prediction5 = get_predictions(keep_types=["label", "category"])
+    actual_items5 = [item.to_dict() for item in actual_prediction5]
+    assert len(actual_items5) == 4
+
+
+def test_image_prediction_collection(client):
+    result = client.simulate_get("/api/v1/images/prediction/collection/")
+    assert result.status_code == 200
+
+    logo_annotation_with_barcode = LogoAnnotationFactory(
+        image_prediction__image__barcode="123"
+    )
+    logo_annotation_with_type = LogoAnnotationFactory(
+        type="label", image_prediction__image__barcode="456"
+    )
+
+    # test with "barcode=123" and "with_logo=True"
+    result = client.simulate_get(
+        "/api/v1/images/prediction/collection",
+        params={
+            "barcode": "123",
+            "with_logo": True,
+        },
+    )
+
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 1
+    assert data["images"][0]["id"] == logo_annotation_with_barcode.id
+    assert data["images"][0]["image_prediction"]["image"]["barcode"] == "123"
+
+    # test with "type=label" and "with_logo=True"
+    result = client.simulate_get(
+        "/api/v1/images/prediction/collection",
+        params={
+            "type": "label",
+            "with_logo": True,
+        },
+    )
+
+    assert result.status_code == 200
+    data = result.json
+    data["images"].sort(key=lambda d: d["id"])
+    assert data["count"] == 2
+    assert data["images"][0]["id"] == logo_annotation_with_barcode.id
+    assert data["images"][1]["id"] == logo_annotation_with_type.id
+
+    # test with "barcode=456" and "with_logo=True"
+    result = client.simulate_get(
+        "/api/v1/images/prediction/collection",
+        params={
+            "barcode": "456",
+            "with_logo": True,
+        },
+    )
+
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 1
+    assert data["images"][0]["id"] == logo_annotation_with_type.id
+
+    # test with "type=label" and "with_logo=False"
+    result = client.simulate_get(
+        "/api/v1/images/prediction/collection",
+        params={
+            "type": "label",
+            "with_logo": False,
+        },
+    )
+
+    assert result.status_code == 200
+    data = result.json
+    assert data["count"] == 0
```

Comment:
If you use an ImagePredictionFactory, you will have a result.

The other tests above look ok to me.